cartographer_ros/cartographer_ros/offline_node_main.cc

```diff
@@ -0,0 +1,190 @@
+/*
+ * Copyright 2016 The Cartographer Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <string>
+#include <vector>
+
+#include "cartographer/common/configuration_file_resolver.h"
+#include "cartographer/common/lua_parameter_dictionary.h"
+#include "cartographer/common/make_unique.h"
+#include "cartographer/common/port.h"
+#include "cartographer_ros/node.h"
+#include "cartographer_ros/ros_log_sink.h"
+#include "cartographer_ros/urdf_reader.h"
+#include "ros/callback_queue.h"
+#include "rosbag/bag.h"
+#include "rosbag/view.h"
+#include "rosgraph_msgs/Clock.h"
+#include "urdf/model.h"
+
+DEFINE_string(configuration_directory, "",
+              "First directory in which configuration files are searched, "
+              "second is always the Cartographer installation to allow "
+              "including files from there.");
+DEFINE_string(configuration_basename, "",
+              "Basename, i.e. not containing any directory prefix, of the "
+              "configuration file.");
+// TODO(damonkohler): Support multi-trajectory.
+DEFINE_string(bag_filename, "", "Bag to process.");
+DEFINE_string(
+    urdf_filename, "",
+    "URDF file that contains static links for your sensor configuration.");
+
+namespace cartographer_ros {
+namespace {
+
+constexpr int kLatestOnlyPublisherQueueSize = 1;
+
+// Default topic names; expected to be remapped as needed.
+constexpr char kLaserScanTopic[] = "scan";
+constexpr char kMultiEchoLaserScanTopic[] = "echoes";
+constexpr char kPointCloud2Topic[] = "points2";
+constexpr char kImuTopic[] = "imu";
```

Comment:
Should we make these already resolved names to make sure that all topics contained in the bag can be remapped without interfering with our default topic names? I.e. Do not call resolve on the `k...Topic` names?